tests/perf/conftest.py

```diff
@@ -15,19 +15,35 @@
 logger = logging.getLogger(__name__)
 
 DEFAULT_ROWS = 100_000_000  # 100 million rows
-PARTITIONS = 10
-
+DEFAULT_PARTITIONS = 10
+DEFAULT_COLUMNS = 4
 SCHEMA_STR = (
     "col1: int, col2: int, col3: int, col4: array<int>, "
     "col5: date, col6: timestamp, col7: map<string, int>, "
     'col8: struct<field1: int>, col9: string, col10: string'
 )
+DEFAULT_BEGIN_DATE = "1900-01-01"
+DEFAULT_END_DATE = "2025-12-31"
+DEFAULT_BEGIN_TIMESTAMP = "1900-01-01 00:00:00"
+DEFAULT_END_TIMESTAMP = "2025-12-31 23:59:59"
+DEFAULT_INTERVAL = "1 second"
 
 REF_SCHEMA_STR = "ref_col1: int, ref_col2: int, ref_col3: int"
 
 RUN_TIME = datetime(2025, 1, 1, 0, 0, 0, 0, tzinfo=timezone.utc)
 
 
+def make_data_gen(
+    spark,
+    n_rows: int = DEFAULT_ROWS,
+    n_columns: int = DEFAULT_COLUMNS,
+    partitions: int = DEFAULT_PARTITIONS,
+):
+    col_names = [f"col{i+1}" for i in range(n_columns)]
+    gen = dg.DataGenerator(spark, rows=n_rows, partitions=partitions)
```

Comment:
it does not make sense to parametrize columns if they are statically defined in generated_df